bsp4j/src/ch/epfl/scala/bsp4j/BuildServer.java

```diff
@@ -81,6 +81,16 @@ public interface BuildServer {
   @JsonRequest("buildTarget/inverseSources")
   CompletableFuture<InverseSourcesResult> buildTargetInverseSources(InverseSourcesParams params);
 
+/**
+ * The wrapped sources request is sent from the client to the server to query for
+ * the list of build targets containing wrapped sources. Wrapped sources are script
+ * sources that are wrapped by the build tool with some top and bottom wrappers.
+ * The server communicates during the initialize handshake whether this method is
+ * supported or not.
+ */
+  @JsonRequest("buildTarget/wrappedSources")
+  CompletableFuture<WrappedSourcesResult> buildTargetWrappedSources(WrappedSourcesParams params);
```

Comment:
Should we actually have a `ScriptBuildServer` as a separate thing?

Reply:
Why should we?
It's a capability that can be declared as supported via normal capabilities. Also, nothing stops someone to write a script wrapper for java which does a similar thing to what scala-cli does.

Reply:
sure, I was just being cautious since we don't have too many server implementing this, but I guess it's totally fine if we add a capability

Reply:
There is probably no clear line to cut between an capability and an extension. The mostly differ in how they get enabled. And this story isn't completely written for modularity. 

@lolgab What stops us to use the `wrapperSources` to use for Java files right away. Is there anything Scala specific?

Reply:
> @lolgab What stops us to use the wrapperSources to use for Java files right away. Is there anything Scala specific?

I don't think anything stops us, but other than the javadoc maybe, there is nothing Scala specific in this PR, unless I'm wrong.

Reply:
Great.